test/cluster/functionality.sh

```diff
@@ -0,0 +1,104 @@
+
+os::test::junit::declare_suite_start "test/cluster/functionality"
+
+os::cmd::expect_success "oc project logging"
+
+# We need to use a name and token for logging checks later,
+# so we have to provision a user with a token for this.
+# TODO: Why is this necessary?
+os::cmd::expect_success "oc login --username=${LOG_ADMIN_USER:-admin} --password=${LOG_ADMIN_PW:-admin}"
+test_user="$( oc whoami )"
+test_token="$( oc whoami -t )"
+os::cmd::expect_success "oc login --username=system:admin"
+
+# We can reach the ElasticSearch service at serviceName:apiPort
+elasticsearch_api="$( oc get svc "${OAL_ELASTICSEACH_SERVICE}" -o jsonpath='{ .metadata.name }:{ .spec.ports[?(@.targetPort=="restapi")].port }' )"
+
+for kibana_pod in $( oc get pods --selector component="${OAL_KIBANA_COMPONENT}"  -o jsonpath='{ .items[*].metadata.name }' ); do
+	os::log::info "Testing Kibana pod ${kibana_pod} for a successful start..."
+	os::cmd::try_until_text "oc logs ${kibana_pod} -c kibana" "Server running at http://0\.0\.0\.0:5601"
+	os::cmd::try_until_text "oc logs ${kibana_pod} -c kibana" "Kibana index ready"
+done
+
+for elasticsearch_pod in $( oc get pods --selector component="${OAL_ELASTICSEACH_COMPONENT}" -o jsonpath='{ .items[*].metadata.name }' ); do
+	os::log::info "Testing ElasticSearch pod ${elasticsearch_pod} for a successful start..."
+	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "\[cluster\.service\s*\]"
+	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "\[node\s*\]\s*\[.*\]\s*started"
+
+	os::log::info "Checking that ElasticSearch pod ${elasticsearch_pod} recovered its indices after starting..."
+	if oc logs "${elasticsearch_pod}" | grep -E "\[cluster\.service\s*\]" | grep -q "new_master"; then
+		os::cmd::expect_success_and_text "oc logs ${elasticsearch_pod}" "\[gateway\s*\]\s*\[.*\]\s*recovered\s*\[[0-9]*\]\s*indices into cluster_state"
+	elif oc logs "${elasticsearch_pod}" | grep -E "\[cluster\.service\s*\]" | grep -q "detected_master"; then
+		os::log::info "ElasticSearch pod ${elasticsearch_pod} was able to detect a master"
+	else
+		os::log::fatal "ElasticSearch pod ${elasticsearch_pod} isn't master and was unable to detect a master"
+	fi
+
+	os::log::info "Checking that ElasticSearch pod ${elasticsearch_pod} contains common data model index templates..."
+	os::cmd::expect_success "oc exec ${elasticsearch_pod} -- ls -1 /usr/share/elasticsearch/index_templates"
+	for template in $( oc exec "${elasticsearch_pod}" -- ls -1 /usr/share/elasticsearch/index_templates ); do
+		os::cmd::expect_success_and_text "oc exec ${elasticsearch_pod} -- curl -sk --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key -X HEAD -w '%{response_code}' https://localhost:9200/_template/${template}" '200'
+	done
+
+	os::log::info "Checking that ElasticSearch pod ${elasticsearch_pod} has persisted indices created by Fluentd..."
+	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "update_mapping \[com\.redhat\.viaq\.common\]"
```

Comment:
Would it be appropriate to make this wait until `/_cat/indices?h=index` returned an index starting with `.operations` or `.project` ?